File: mobile/flutter/ascend_flutter/lib/features/posts/data/repositories/posts_repository_impl.dart
import '../../domain/entities/post.dart';
import '../../domain/repositories/posts_repository.dart';
import '../datasources/posts_remote_data_source.dart';

class PostsRepositoryImpl implements PostsRepository {
  final PostsRemoteDataSource remoteDataSource;

  const PostsRepositoryImpl(this.remoteDataSource);

  @override
  Future<List<Post>> getPosts() {
    return remoteDataSource.getPosts();
  }
}

File: mobile/react-native/ascend_rn/src/features/posts/data/repositories/PostsRepositoryImpl.ts
import { Post } from '../../domain/entities/Post';
import { PostsRepository } from '../../domain/repositories/PostsRepository';
import { PostsRemoteDataSource } from '../datasources/PostsRemoteDataSource';

export class PostsRepositoryImpl
  implements PostsRepository {

  constructor(
    private readonly remoteDataSource: PostsRemoteDataSource,
  ) {}

  async getPosts(): Promise<Post[]> {
    return this.remoteDataSource.getPosts();
  }
}